### PR TITLE
follow me example updates for dkpy2

### PR DESCRIPTION
**Pull request by hamishwillee**
Updated the Follow me example to DKPY2 and to use the same "pattern" as other docs and examples.

As a side note, I don't think this would have ever worked prior to this update because there was no code to get the vehicle to take off, and because one of the checks on whether or not to run the loop to get GPS would never have run. 

This was tested in a simulated environment using simulated gps :-)

### CODE REVIEW

File: docs/examples/follow_me.rst [hunk 1]
```diff
@@ -4,39 +4,155 @@
...
+                print "User has changed flight modes - aborting follow-me"
+                break    
+                
+            # Read the GPS state from the laptop
+            gpsd.next()
+
+            # Once we have a valid location (see gpsd documentation) we can start moving our vehicle around
+            if (gpsd.valid & gps.LATLON_SET) != 0:
+                altitude = 30  # in meters
+                dest = Location(gpsd.fix.latitude, gpsd.fix.longitude, altitude, is_relative=True)
```
Comment (tcr3dr) [inline docs/examples/follow_me.rst]: I believe Location will no longer work on master. This is now `LocationGlobal` i beliece?

File: docs/examples/follow_me.rst [hunk 2]
```diff
@@ -4,39 +4,155 @@
...
+    import gps
+    import socket
+    
+    ...
+
+    try:
+        # Use the python gps package to access the laptop GPS
+        gpsd = gps.gps(mode=gps.WATCH_ENABLE)
+
+        #Arm and take of to altitude of 5 meters
```
Comment (tcr3dr) [inline docs/examples/follow_me.rst]: off spelled wrong, missing space after #

File: docs/examples/follow_me.rst [hunk 3]
```diff
@@ -4,39 +4,155 @@
...
+                # A better implementation would only send new waypoints if the position had changed significantly
+                vehicle.commands.goto(dest)
+                vehicle.flush()
+
+                # Send a new target every two seconds
+                # For a complete implementation of follow me you'd want adjust this delay
+                time.sleep(2)
+                
+    except socket.error:
+        print "Error: gpsd service does not seem to be running, plug in USB GPS or run run-fake-gps.sh"
```
Comment (tcr3dr) [inline docs/examples/follow_me.rst]: might want to sys.exit(1) after this

File: docs/examples/follow_me.rst (hunk 1 above)
Comment (tcr3dr) [inline docs/examples/follow_me.rst]: (We should probably re-export LocationGlobal as Location for backcompat though...)

File: examples/follow_me/follow_me.py [hunk 4]
```diff
@@ -1,65 +1,109 @@
...
+            print "User has changed flight modes - aborting follow-me"
+            break    
+            
+        # Read the GPS state from the laptop
+        gpsd.next()
+
+        # Once we have a valid location (see gpsd documentation) we can start moving our vehicle around
+        if (gpsd.valid & gps.LATLON_SET) != 0:
+            altitude = 30  # in meters
+            dest = Location(gpsd.fix.latitude, gpsd.fix.longitude, altitude, is_relative=True)
```
Comment (tcr3dr) [inline examples/follow_me/follow_me.py]: Same here

File: examples/follow_me/follow_me.py [hunk 5]
```diff
@@ -1,65 +1,109 @@
...
+            # A better implementation would only send new waypoints if the position had changed significantly
+            vehicle.commands.goto(dest)
+            vehicle.flush()
+
+            # Send a new target every two seconds
+            # For a complete implementation of follow me you'd want adjust this delay
+            time.sleep(2)
+            
+except socket.error:
+    print "Error: gpsd service does not seem to be running, plug in USB GPS or run run-fake-gps.sh"
```
Comment (tcr3dr) [inline examples/follow_me/follow_me.py]: Same as above

File: docs/examples/follow_me.rst (hunk 1 above)
Comment (hamishwillee) [inline docs/examples/follow_me.rst]: @tcr3dr YOu believe wrong - see #351 is still unmerged. I'll update this when that goes in.

File: examples/follow_me/follow_me.py (hunk 4 above)
Comment (hamishwillee) [inline examples/follow_me/follow_me.py]: Nope, location not yet merged.

File: docs/examples/follow_me.rst (hunk 3 above)
Comment (hamishwillee) [inline docs/examples/follow_me.rst]: Good idea

**Comment by hamishwillee** [comment]
@tcr3dr I added fixes to your suggestions. The new location object has not yet been merged. Sebastian did want it ASAP and asked for this on the associated pull request.

Please merge this in (or help walk me through what I'm doing wrong with these cursed rebases)

**Comment by tcr3dr** [comment]
@hamishwillee A good strategy is `git rebase origin/master -X theirs` which will attempt to resolve commits for you, to a point.

**Comment by hamishwillee** [comment]
Thanks Tim, I'll try that rebase strategy next time.

**Pull request merged**